airflow/utils/cli.py

```diff
@@ -269,7 +269,7 @@ def get_dag_by_pickle(pickle_id: int, session: Session = NEW_SESSION) -> DAG:
     """Fetch DAG from the database using pickling."""
     from airflow.models import DagPickle
 
-    dag_pickle = session.scalar(select(DagPickle).where(DagPickle.id == pickle_id)).first()
+    dag_pickle = session.scalar(select(DagPickle).where(DagPickle.id == pickle_id).limit(1))
```

Comment:
If one of the errors you saw was `AttributeError: None object has no attribute 'first'`, then this change could still produce that error, as long as `session.scalar(select(DagPickle).where(DagPickle.id == pickle_id)` returns `None`. Maybe it would be better to check for `None` before calling something on it

Reply:
The output of `session.scalar(select(DagPickle).where(DagPickle.id == pickle_id)` could be DagPickle object in case of requested id exists, or None if it not exists, both of this object doesn't have method `first()`

Most probably the issue happen when this function changed to new SQLALchemy API instead of legacy Query API.

Reply:
But if no DagPickle object is found, then calling the `limit()` method will still throw an Exception

Reply:
And this not tested before, so I was add test case for both: pickle_id exists or not

Reply:
> But if no DagPickle object is found, then calling the limit() method will still throw an Exception

Nope, this one is transformed to a single query `select(DagPickle).where(DagPickle.id == pickle_id).limit(1)`

Which would transformed to something like

```sql
SELECT *
FROM dag_pickle
WHERE id = 42
LIMIT 1
```

Reply:
This could be run without binding session, that is a bigger benefit of "new" SA API

```python
from sqlalchemy import select

from airflow.models import DagPickle

stmt = select(DagPickle).where(DagPickle.id == 42).limit(1)
print(stmt)
```

```console
SELECT dag_pickle.id, dag_pickle.pickle, dag_pickle.created_dttm, dag_pickle.pickle_hash 
FROM dag_pickle 
WHERE dag_pickle.id = :id_1
 LIMIT :param_1
```

Reply:
Ah wow, that's neat. Cool, looks good to me.